library/core/src/macros/mod.rs

```diff
@@ -1273,6 +1273,19 @@ pub(crate) mod builtin {
     /// first macro invocation leading up to the invocation of the `file!`
     /// macro.
     ///
+    /// The file name is derived from the root module's source path passed to the Rust compiler
+    /// and the sequence the compiler takes to get from root module to the
```

Comment:
This would then naturally be
```suggestion
    /// and the sequence the compiler takes to get from the crate root to the
```